src/dcat-ap/dcat-dataset.ts

```diff
@@ -0,0 +1,120 @@
+import { getUserUrl, IItem } from '@esri/arcgis-rest-portal';
+import { localeToLang } from './dcat-formatters';
+import * as _ from 'lodash';
+import { isPage } from '@esri/hub-sites';
+import { UserSession } from '@esri/arcgis-rest-auth';
+
+export class DcatDataset {
+  //  static requiredPaths = [
+  //    'default.id',
+  //    'default.url',
+  //    'item.owner',
+  //    'item.title',
+  //    'item.tags',
+  //    'item.description',
+  //    'item.culture',
+  //    'item.created',
+  //    'metadata.metadata',
+  //    'layer.geometryType',
+  //    'server.supportedExtensions'
+  //  ];
+
+  private _dto: any;
+  private _portalUrl: string;
+  private _orgTitle: string;
+  private _siteUrl: string;
+
+  constructor (dto: any, portalUrl: string, orgTitle: string, siteUrl: string) {
+    this._dto = dto;
+    this._portalUrl = portalUrl;
+    this._orgTitle = orgTitle;
+    this._siteUrl = siteUrl;
+  }
+
+  /**
+    * Lookup fn
+    */
+  private _get (path) {
+    return _.get(this._dto, path, null);
+  }
+
+  get id (): string { return this._get('id'); }
+  get url (): string { return this._get('url'); }
+  get landingPage (): string { return `${this._siteUrl}/datasets/${this.id}`; }
+  get title (): string { return this._get('name'); }
+  get description (): string { return this._get('description'); }
+  get owner (): string { return this._get('owner'); }
+  get ownerUri (): string {
+    return getUserUrl({
+      portal: `${this._portalUrl}/sharing/rest`,
+      username: this.owner
+    } as UserSession) + '?f=json';
+  }
+
+  get language (): string {
+    return this._metaLanguage || localeToLang(this._get('culture')) || null;
+  }
+
+  get keyword (): string[] {
+    if (this._metaKeyword) {
+      return this._metaKeyword;
+    }
+
+    let tags = this._get('tags');
+
+    const hasNoTags = !tags || tags.length === 0 || !tags[0]; // if tags is undefined, the tags array is empty, or tags is an empty string
+    if (isPage({
+      type: this._dto.type,
+      typeKeywords: this._dto.typeKeywords
+    } as IItem) && hasNoTags) {
+      tags = ['ArcGIS Hub page'];
+    }
+
+    return tags;
+  }
+
+  /**
+    * Returns an ISO string
+    */
+  get issuedDateTime (): string {
+    return this._metaPubDate || new Date(this._get('created')).toISOString();
+  }
+
+  get isFeatureLayer (): boolean { return /_/.test(this.id); }
+  get hasGeometryType (): boolean { return !!this._get('geometryType'); }
+  private get _supportedExtensions () { return _.get(this._dto, 'server.supportedExtensions'); }
+  get supportsWFS (): boolean { return this._supportedExtensions && this._supportedExtensions.includes('WFSServer'); }
+  get supportsWMS (): boolean { return this._supportedExtensions && this._supportedExtensions.includes('WMSServer'); }
```

Comment:
These will always be false until we expose `supportedExtensions` in some form through the Search API